src/rust/ide/view/src/documentation.rs

```diff
@@ -133,22 +148,42 @@ impl ViewModel {
             //              https://github.com/enso-org/enso/issues/1063
             let processed = string.replace("\\n", "\n").replace("\"", "");
             let output = match input_type {
-                InputFormat::AST  => parser.generate_html_docs(processed),
+                InputFormat::AST       => parser.generate_html_docs(processed),
                 InputFormat::Docstring => parser.generate_html_doc_pure(processed),
             };
             let output = output?;
-            if output.is_empty() {
-                Ok(PLACEHOLDER_STR.into())
-            } else {
-                Ok(output)
-            }
+            if output.is_empty() { Ok(PLACEHOLDER_STR.into()) }
+            else                 { Ok(output)                 }
         }
     }
 
     /// Create a container for generated content and embed it with stylesheet.
     fn push_to_dom(&self, content:String) {
         let data_str = format!(r#"<div class="docVis">{}{}</div>"#,documentation_style(),content);
-        self.dom.dom().set_inner_html(&data_str)
+        self.dom.dom().set_inner_html(&data_str);
+    }
+
+    /// Append listeners to copy buttons in doc to enable copying examples.
+    fn set_listeners_to_copy_buttons(&self) {
+        let code_blocks  = self.dom.dom().get_elements_by_class_name("CodeBlock");
+        let copy_buttons = self.dom.dom().get_elements_by_class_name("copyCodeBtn");
+        let closures     = (0..copy_buttons.length()).map(|i| -> OptClosure {
+            let copy_button = copy_buttons.get_with_index(i)?.dyn_into::<HtmlElement>().ok()?;
+            let code_block  = code_blocks.get_with_index(i)?.dyn_into::<HtmlElement>().ok()?;
```

Comment:
there is no error handling here. Do we consider a situation when we have a code block but no copy button ok? I believe that this should be reported as an error, otherwise, we will have some day documentation not working without any errors nor warnings and we will debug it for a long tie before we will find these magical question marks here.

Reply:
As I wrote in my previous review, after mapping to the Option<Closure> we should do the filter_map with logging error if element is None.

Reply:
This is still not done.